### DISCUSSION TITLE
Get weight of particular Layer

### DISCUSSION THREAD

**Discussion by atale480** [pyg-team/pytorch_geometric#6511] [Q&A] [answered]
Considering code below how can I get weight of any layer? Because I want to save weight during training.
```
import torch
import torch.nn.functional as F
from torch_geometric.nn import GCNConv

class Net(torch.nn.Module):
    def __init__(self):
        super(Net, self).__init__()
        self.conv1 = GCNConv(dataset.num_node_features, 16)
        self.conv2 = GCNConv(16, dataset.num_classes)

    def forward(self, data):
        x, edge_index = data.x, data.edge_index

        x = self.conv1(x, edge_index)
        x = F.relu(x)
        x = F.dropout(x, training=self.training)
        x = self.conv2(x, edge_index)

        return F.log_softmax(x, dim=1)
```

Here suppose I want to get weight of layer `conv2` or `conv1` .
How can I get those?

I was trying to use `register_hook` from PyTorch. but As I don't know how to get weights I'm stuck here.

**Comment by wsad1** [comment, marked as answer]
A `hook` is used to get intermediate output of some layer in your neural network, not to get the wieghts. To get the weights of lets say `conv2` simply do
```
net = Net()
print(net.conv2.lin.weight, net.conv2.bias.weight)
```

**Reply by atale480** [reply]
what is `.lin`  in `net.conv2.lin.weight`?

**Reply by wsad1** [reply]
Its the linear transform applied to each node feature.
In other words, it is the `theta` in https://pytorch-geometric.readthedocs.io/en/latest/modules/nn.html#torch_geometric.nn.conv.GCNConv.
